adapters/play/src/main/scala/caliban/PlayAdapter.scala

```diff
@@ -70,15 +83,28 @@ trait PlayAdapter[R] {
     interpreter: GraphQLInterpreter[R, E],
     skipValidation: Boolean = false,
     enableIntrospection: Boolean = true
-  )(implicit runtime: Runtime[R]): Action[GraphQLRequest] =
-    actionBuilder.async(parse.json[GraphQLRequest])(req =>
-      executeRequest(
-        interpreter,
-        req,
-        skipValidation,
-        enableIntrospection
-      )
-    )
+  )(implicit runtime: Runtime[R]): Action[AnyContent] = actionBuilder.async { req =>
+    req.contentType match {
+      case Some(`application/graphql`) =>
+        executeRequest(
+          interpreter,
+          req.withBody(GraphQLRequest(query = req.body.asText)),
+          skipValidation,
+          enableIntrospection
+        )
+      case _ =>
+        Json.fromJson[GraphQLRequest](req.body.asJson.getOrElse(JsString("Empty request body"))) match {
```

Comment:
Any reason not to use `parse.json[GraphQLRequest])` anymore? It was more concise.

Reply:
Thanks for reviewing, I could not manage to use it because it returns a BodyParser[GraphQLRequest] and I couldn't build the request body from it

Reply:
You can create a `BodyParser` that looks like this:

```scala
private def graphqlBodyParser(implicit ec: ExecutionContext): BodyParser[GraphQLRequest] = parse.using { rh =>
  rh.contentType match {
    case Some(`application/graphql`) => parse.text.map(text => GraphQLRequest(query = Some(text))
    case _ => parse.json[GraphQLRequest]
  }
}
```